# Day 1: Data Cleaning

In [77]:
import pandas as pd

In [78]:
despesas_2019 = pd.read_csv('despesa_ceaps_2019.csv', encoding="ISO-8859-1",delimiter=';')
despesas_2020 = pd.read_csv('despesa_ceaps_2020.csv', encoding="ISO-8859-1",delimiter=';')
despesas_2021 = pd.read_csv('despesa_ceaps_2021.csv', encoding="ISO-8859-1",delimiter=';')
despesas_2022 = pd.read_csv('despesa_ceaps_2022.csv', encoding="ISO-8859-1",delimiter=';')

In [79]:
# Concatenate them
combined_df = pd.concat([despesas_2019, despesas_2020, despesas_2021, despesas_2022])

# Reset the index if you want a new continuous index
combined_df.reset_index(drop=True, inplace=True)


In [80]:
combined_df

ANO  MES           SENADOR  \
0      2019    1      ACIR GURGACZ   
1      2019    1      ACIR GURGACZ   
2      2019    1      ACIR GURGACZ   
3      2019    1      ACIR GURGACZ   
4      2019    2      ACIR GURGACZ   
...     ...  ...               ...   
69328  2022   12  ZEQUINHA MARINHO   
69329  2022   12  ZEQUINHA MARINHO   
69330  2022   12  ZEQUINHA MARINHO   
69331  2022   12  ZEQUINHA MARINHO   
69332  2022   12  ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
0      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
1      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
2      Aluguel de im�veis para escrit�rio pol�t...      004.948.028-63   
3      Aluguel de im�veis para escrit�rio pol�t...  05.423.963/0001-11   
4      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                          FORNECEDOR  DOCUMENTO        DATA  \
0                           ENERGISA  006582758  04/01/2019   
1                           ENERGISA  006582755  04/01/2019   
2      GILBERTO PISELO DO NASCIMENTO      00119  07/01/2019   
3                    OI M�VEL S.A.   86161151  25/12/2018   
4                           ENERGISA  007236036  04/02/2019   
...                              ...        ...         ...   
69328                  Exceller Tour     WIXHAI  06/12/2022   
69329                  Exceller Tour     WITOLM  09/12/2022   
69330                  Exceller Tour     THPKVQ  20/12/2022   
69331                  Exceller Tour     QNN9HX  21/12/2022   
69332                  Exceller Tour     WMQWBX  30/12/2022   

                                            DETALHAMENTO VALOR_REEMBOLSADO  \
0      Despesa com pagamento de energia el�trica do...             66,02   
1      Despesa com pagamento de energia el�trica do...            139,98   
2      Despesa com pagamento de aluguel de im�vel p...              6000   
3      Despesa com pagamento de telefonia para o escr...            316,39   
4      Despesa com pagamento de energia el�trica pa...             99,45   
...                                                  ...               ...   
69328  Companhia A�rea: LATAM, Localizador: WIXHAI....           2893,04   
69329  Companhia A�rea: GOL, Localizador: WITOLM. P...           1180,19   
69330  Companhia A�rea: TAM, Localizador: THPKVQ. P...            2671,9   
69331  Companhia A�rea: AZUL, Localizador: QNN9HX. ...           1334,31   
69332  Companhia A�rea: TAM, Localizador: WMQWBX. P...           2250,72   

       COD_DOCUMENTO  
0            2116543  
1            2116546  
2            2113817  
3            2116541  
4            2116550  
...              ...  
69328        2191398  
69329        2192272  
69330        2192274  
69331        2192244  
69332        2193622  

[69333 rows x 11 columns]

In [81]:
combined_df.drop_duplicates()

ANO  MES           SENADOR  \
0      2019    1      ACIR GURGACZ   
1      2019    1      ACIR GURGACZ   
2      2019    1      ACIR GURGACZ   
3      2019    1      ACIR GURGACZ   
4      2019    2      ACIR GURGACZ   
...     ...  ...               ...   
69328  2022   12  ZEQUINHA MARINHO   
69329  2022   12  ZEQUINHA MARINHO   
69330  2022   12  ZEQUINHA MARINHO   
69331  2022   12  ZEQUINHA MARINHO   
69332  2022   12  ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
0      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
1      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
2      Aluguel de im�veis para escrit�rio pol�t...      004.948.028-63   
3      Aluguel de im�veis para escrit�rio pol�t...  05.423.963/0001-11   
4      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                          FORNECEDOR  DOCUMENTO        DATA  \
0                           ENERGISA  006582758  04/01/2019   
1                           ENERGISA  006582755  04/01/2019   
2      GILBERTO PISELO DO NASCIMENTO      00119  07/01/2019   
3                    OI M�VEL S.A.   86161151  25/12/2018   
4                           ENERGISA  007236036  04/02/2019   
...                              ...        ...         ...   
69328                  Exceller Tour     WIXHAI  06/12/2022   
69329                  Exceller Tour     WITOLM  09/12/2022   
69330                  Exceller Tour     THPKVQ  20/12/2022   
69331                  Exceller Tour     QNN9HX  21/12/2022   
69332                  Exceller Tour     WMQWBX  30/12/2022   

                                            DETALHAMENTO VALOR_REEMBOLSADO  \
0      Despesa com pagamento de energia el�trica do...             66,02   
1      Despesa com pagamento de energia el�trica do...            139,98   
2      Despesa com pagamento de aluguel de im�vel p...              6000   
3      Despesa com pagamento de telefonia para o escr...            316,39   
4      Despesa com pagamento de energia el�trica pa...             99,45   
...                                                  ...               ...   
69328  Companhia A�rea: LATAM, Localizador: WIXHAI....           2893,04   
69329  Companhia A�rea: GOL, Localizador: WITOLM. P...           1180,19   
69330  Companhia A�rea: TAM, Localizador: THPKVQ. P...            2671,9   
69331  Companhia A�rea: AZUL, Localizador: QNN9HX. ...           1334,31   
69332  Companhia A�rea: TAM, Localizador: WMQWBX. P...           2250,72   

       COD_DOCUMENTO  
0            2116543  
1            2116546  
2            2113817  
3            2116541  
4            2116550  
...              ...  
69328        2191398  
69329        2192272  
69330        2192274  
69331        2192244  
69332        2193622  

[69333 rows x 11 columns]

In [82]:
combined_df.dropna()

ANO  MES           SENADOR  \
0      2019    1      ACIR GURGACZ   
1      2019    1      ACIR GURGACZ   
2      2019    1      ACIR GURGACZ   
3      2019    1      ACIR GURGACZ   
4      2019    2      ACIR GURGACZ   
...     ...  ...               ...   
69328  2022   12  ZEQUINHA MARINHO   
69329  2022   12  ZEQUINHA MARINHO   
69330  2022   12  ZEQUINHA MARINHO   
69331  2022   12  ZEQUINHA MARINHO   
69332  2022   12  ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
0      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
1      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
2      Aluguel de im�veis para escrit�rio pol�t...      004.948.028-63   
3      Aluguel de im�veis para escrit�rio pol�t...  05.423.963/0001-11   
4      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                          FORNECEDOR  DOCUMENTO        DATA  \
0                           ENERGISA  006582758  04/01/2019   
1                           ENERGISA  006582755  04/01/2019   
2      GILBERTO PISELO DO NASCIMENTO      00119  07/01/2019   
3                    OI M�VEL S.A.   86161151  25/12/2018   
4                           ENERGISA  007236036  04/02/2019   
...                              ...        ...         ...   
69328                  Exceller Tour     WIXHAI  06/12/2022   
69329                  Exceller Tour     WITOLM  09/12/2022   
69330                  Exceller Tour     THPKVQ  20/12/2022   
69331                  Exceller Tour     QNN9HX  21/12/2022   
69332                  Exceller Tour     WMQWBX  30/12/2022   

                                            DETALHAMENTO VALOR_REEMBOLSADO  \
0      Despesa com pagamento de energia el�trica do...             66,02   
1      Despesa com pagamento de energia el�trica do...            139,98   
2      Despesa com pagamento de aluguel de im�vel p...              6000   
3      Despesa com pagamento de telefonia para o escr...            316,39   
4      Despesa com pagamento de energia el�trica pa...             99,45   
...                                                  ...               ...   
69328  Companhia A�rea: LATAM, Localizador: WIXHAI....           2893,04   
69329  Companhia A�rea: GOL, Localizador: WITOLM. P...           1180,19   
69330  Companhia A�rea: TAM, Localizador: THPKVQ. P...            2671,9   
69331  Companhia A�rea: AZUL, Localizador: QNN9HX. ...           1334,31   
69332  Companhia A�rea: TAM, Localizador: WMQWBX. P...           2250,72   

       COD_DOCUMENTO  
0            2116543  
1            2116546  
2            2113817  
3            2116541  
4            2116550  
...              ...  
69328        2191398  
69329        2192272  
69330        2192274  
69331        2192244  
69332        2193622  

[36752 rows x 11 columns]

In [83]:
# Substituir vírgulas por pontos na coluna 'VALOR_REEMBOLSADO'
combined_df['VALOR_REEMBOLSADO'] = combined_df['VALOR_REEMBOLSADO'].str.replace(',', '.')

# Convertendo a coluna 'VALOR_REEMBOLSADO' para tipo numérico (float)
combined_df['VALOR_REEMBOLSADO'] = combined_df['VALOR_REEMBOLSADO'].astype(float)



In [84]:
combined_df

ANO  MES           SENADOR  \
0      2019    1      ACIR GURGACZ   
1      2019    1      ACIR GURGACZ   
2      2019    1      ACIR GURGACZ   
3      2019    1      ACIR GURGACZ   
4      2019    2      ACIR GURGACZ   
...     ...  ...               ...   
69328  2022   12  ZEQUINHA MARINHO   
69329  2022   12  ZEQUINHA MARINHO   
69330  2022   12  ZEQUINHA MARINHO   
69331  2022   12  ZEQUINHA MARINHO   
69332  2022   12  ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
0      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
1      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
2      Aluguel de im�veis para escrit�rio pol�t...      004.948.028-63   
3      Aluguel de im�veis para escrit�rio pol�t...  05.423.963/0001-11   
4      Aluguel de im�veis para escrit�rio pol�t...  05.914.650/0001-66   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                          FORNECEDOR  DOCUMENTO        DATA  \
0                           ENERGISA  006582758  04/01/2019   
1                           ENERGISA  006582755  04/01/2019   
2      GILBERTO PISELO DO NASCIMENTO      00119  07/01/2019   
3                    OI M�VEL S.A.   86161151  25/12/2018   
4                           ENERGISA  007236036  04/02/2019   
...                              ...        ...         ...   
69328                  Exceller Tour     WIXHAI  06/12/2022   
69329                  Exceller Tour     WITOLM  09/12/2022   
69330                  Exceller Tour     THPKVQ  20/12/2022   
69331                  Exceller Tour     QNN9HX  21/12/2022   
69332                  Exceller Tour     WMQWBX  30/12/2022   

                                            DETALHAMENTO  VALOR_REEMBOLSADO  \
0      Despesa com pagamento de energia el�trica do...              66.02   
1      Despesa com pagamento de energia el�trica do...             139.98   
2      Despesa com pagamento de aluguel de im�vel p...            6000.00   
3      Despesa com pagamento de telefonia para o escr...             316.39   
4      Despesa com pagamento de energia el�trica pa...              99.45   
...                                                  ...                ...   
69328  Companhia A�rea: LATAM, Localizador: WIXHAI....            2893.04   
69329  Companhia A�rea: GOL, Localizador: WITOLM. P...            1180.19   
69330  Companhia A�rea: TAM, Localizador: THPKVQ. P...            2671.90   
69331  Companhia A�rea: AZUL, Localizador: QNN9HX. ...            1334.31   
69332  Companhia A�rea: TAM, Localizador: WMQWBX. P...            2250.72   

       COD_DOCUMENTO  
0            2116543  
1            2116546  
2            2113817  
3            2116541  
4            2116550  
...              ...  
69328        2191398  
69329        2192272  
69330        2192274  
69331        2192244  
69332        2193622  

[69333 rows x 11 columns]

# Day 2: Data Visualization & Storytelling

In [85]:
combined_df.columns

Index(['ANO', 'MES', 'SENADOR', 'TIPO_DESPESA', 'CNPJ_CPF', 'FORNECEDOR',
       'DOCUMENTO', 'DATA', 'DETALHAMENTO', 'VALOR_REEMBOLSADO',
       'COD_DOCUMENTO'],
      dtype='object')

In [86]:
combined_df.describe()

ANO           MES  VALOR_REEMBOLSADO  COD_DOCUMENTO
count  69333.000000  69333.000000       69333.000000   6.933300e+04
mean    2020.414766      6.584945        1413.131098   2.153664e+06
std        1.162288      3.369021        3098.174309   2.257438e+04
min     2019.000000      1.000000           0.010000   2.112846e+06
25%     2019.000000      4.000000         158.140000   2.134374e+06
50%     2020.000000      7.000000         426.040000   2.153693e+06
75%     2021.000000     10.000000        1570.560000   2.172904e+06
max     2022.000000     12.000000      120000.000000   2.198297e+06

In [87]:
pip install plotly

You should consider upgrading via the 'c:\Users\camil\AppData\Local\Programs\Python\Python310\python.exe -m pip install --upgrade pip' command.


In [88]:
import plotly.offline as py
import plotly.graph_objs as go

### Qual foi o senador que mais declarou despesas?

In [89]:
total_por_senador = combined_df.groupby('SENADOR')['VALOR_REEMBOLSADO'].sum()

# Ordenar os resultados em ordem decrescente
total_por_senador = total_por_senador.sort_values(ascending=False)

# Obter o senador que mais declarou despesas
senador_mais_despesas = total_por_senador.idxmax()

print("O senador que mais declarou despesas é:", senador_mais_despesas)


O senador que mais declarou despesas é: TELM�RIO MOTA


In [90]:
trace = go.Bar(
    x=total_por_senador.index,  # Senadores no eixo x
    y=total_por_senador.values,  # Valores reembolsados no eixo y
)

data = [trace]

layout = go.Layout(
    title='Despesas Reembolsadas por Senador',
    xaxis=dict(title='Senador'),
    yaxis=dict(title='Valor Reembolsado (R$)'),
)

fig = go.Figure(data=data, layout=layout)

fig.show()

### Qual foi o percentual da cota que os senadores gastaram? 


In [91]:
total_cota = combined_df['VALOR_REEMBOLSADO'].sum()

# Calcula o percentual da cota gasto por senador
percentual_por_senador = (total_por_senador / total_cota) * 100

trace = go.Bar(
    x=percentual_por_senador.index,   # Senadores no eixo x
    y=percentual_por_senador.values,  # Percentuais no eixo y
    marker=dict(color='skyblue'),     # Cor das barras
)

data = [trace]

layout = go.Layout(
    title='Percentual da Cota que os Senadores Gastaram',
    xaxis=dict(title='Senador'),           # Título do eixo x
    yaxis=dict(title='Percentual da Cota'), # Título do eixo y
)

fig = go.Figure(data=data, layout=layout)

fig.show()

### Média dos Gastos dos Senadores por Ano

In [92]:
# Converte a coluna 'DATA' para o formato de data adequado
combined_df['DATA'] = pd.to_datetime(combined_df['DATA'], format='%d/%m/%Y', errors='coerce')

# Verifica se há valores incorretos após a correção
print(combined_df['DATA'].loc[combined_df['DATA'].isnull()])

10986   NaT
31996   NaT
37872   NaT
Name: DATA, dtype: datetime64[ns]


In [93]:
combined_df['ANO'] = pd.to_datetime(combined_df['DATA'], format='%d/%m/%Y').dt.year

# Agrupar os gastos por ano e calcular a média dos valores reembolsados em cada ano
media_por_ano = combined_df.groupby('ANO')['VALOR_REEMBOLSADO'].mean()

trace = go.Scatter(
    x=media_por_ano.index,    # Ano no eixo x
    y=media_por_ano.values,   # Média dos valores reembolsados no eixo y
    mode='lines+markers',     # Tipo de gráfico (linhas com marcadores)
    marker=dict(color='blue'),# Cor da linha
)

data = [trace]

layout = go.Layout(
    title='Média dos Gastos dos Senadores por Ano',
    xaxis=dict(title='Ano'),                # Título do eixo x
    yaxis=dict(title='Média Reembolsada'),  # Título do eixo y
)

fig = go.Figure(data=data, layout=layout)

fig.show()

# Day 3: Forecasting
### Técnica 1: Mais básica: Média dos últimos 3 meses

In [94]:
combined_df['MES'] = pd.to_datetime(combined_df['DATA'], format='%d/%m/%Y').dt.month

In [95]:
df_last_year = combined_df.loc[combined_df['ANO'] == 2022.0]
df_last_year

ANO   MES            SENADOR  \
36108  2022.0   1.0  ALESSANDRO VIEIRA   
36181  2022.0   2.0     ANGELO CORONEL   
36202  2022.0   3.0     ANGELO CORONEL   
36212  2022.0   3.0     ANGELO CORONEL   
36222  2022.0   3.0     ANGELO CORONEL   
...       ...   ...                ...   
69328  2022.0  12.0   ZEQUINHA MARINHO   
69329  2022.0  12.0   ZEQUINHA MARINHO   
69330  2022.0  12.0   ZEQUINHA MARINHO   
69331  2022.0  12.0   ZEQUINHA MARINHO   
69332  2022.0  12.0   ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
36108  Contrata��o de consultorias, assessorias, ...  23.412.247/0001-10   
36181            Divulga��o da atividade parlamentar  07.053.147/0001-99   
36202  Locomo��o, hospedagem, alimenta��o, co...  13.331.488/0001-81   
36212  Locomo��o, hospedagem, alimenta��o, co...  13.331.488/0001-81   
36222  Contrata��o de consultorias, assessorias, ...  30.727.829/0001-51   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                                              FORNECEDOR        DOCUMENTO  \
36108                    AMAZON AWS SERVICOS BRASIL LTDA         00665692   
36181  Velip - Tradeal Servi�os em Tecnologia da In...         00016356   
36202                         J. Pedro de Brito - Carros  202200000000001   
36212                         J. Pedro de Brito - Carros  202200000000002   
36222                    BUNKER BRAIN AUDIOVISUAL EIRELI               79   
...                                                  ...              ...   
69328                                      Exceller Tour           WIXHAI   
69329                                      Exceller Tour           WITOLM   
69330                                      Exceller Tour           THPKVQ   
69331                                      Exceller Tour           QNN9HX   
69332                                      Exceller Tour           WMQWBX   

            DATA                                       DETALHAMENTO  \
36108 2022-01-03                                                NaN   
36181 2022-02-14  Distribui��o de conte�do - divulga��...   
36202 2022-03-05  Loca��o de ve�culo de apoio � atividad...   
36212 2022-03-05  Loca��o de ve�culo de apoio � atividad...   
36222 2022-03-15  Assessoria de comunica��o - planejamento e...   
...          ...                                                ...   
69328 2022-12-06  Companhia A�rea: LATAM, Localizador: WIXHAI....   
69329 2022-12-09  Companhia A�rea: GOL, Localizador: WITOLM. P...   
69330 2022-12-20  Companhia A�rea: TAM, Localizador: THPKVQ. P...   
69331 2022-12-21  Companhia A�rea: AZUL, Localizador: QNN9HX. ...   
69332 2022-12-30  Companhia A�rea: TAM, Localizador: WMQWBX. P...   

       VALOR_REEMBOLSADO  COD_DOCUMENTO  
36108             116.25        2173827  
36181           10000.00        2174668  
36202            4750.00        2175861  
36212            4750.00        2175864  
36222           48000.00        2176306  
...                  ...            ...  
69328            2893.04        2191398  
69329            1180.19        2192272  
69330            2671.90        2192274  
69331            1334.31        2192244  
69332            2250.72        2193622  

[16733 rows x 11 columns]

In [96]:
three_months = [12.0,11.0,10.0]

In [97]:
df_last_three_months = df_last_year.loc[df_last_year['MES'].isin(three_months)]

In [98]:
df_last_three_months

ANO   MES           SENADOR  \
39502  2022.0  10.0   ELIANE NOGUEIRA   
51961  2022.0  12.0    WEVERTON ROCHA   
52652  2022.0  10.0      ACIR GURGACZ   
52653  2022.0  10.0      ACIR GURGACZ   
52654  2022.0  10.0      ACIR GURGACZ   
...       ...   ...               ...   
69328  2022.0  12.0  ZEQUINHA MARINHO   
69329  2022.0  12.0  ZEQUINHA MARINHO   
69330  2022.0  12.0  ZEQUINHA MARINHO   
69331  2022.0  12.0  ZEQUINHA MARINHO   
69332  2022.0  12.0  ZEQUINHA MARINHO   

                                            TIPO_DESPESA            CNPJ_CPF  \
39502  Locomo��o, hospedagem, alimenta��o, co...  34.988.039/0001-80   
51961  Locomo��o, hospedagem, alimenta��o, co...  32.965.877/0001-03   
52652  Aluguel de im�veis para escrit�rio pol�t...      004.948.028-63   
52653  Locomo��o, hospedagem, alimenta��o, co...  34.274.233/0097-46   
52654  Passagens a�reas, aqu�ticas e terrestres n...  17.872.428/0001-27   
...                                                  ...                 ...   
69328  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69329  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69330  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69331  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   
69332  Passagens a�reas, aqu�ticas e terrestres n...  22.052.777/0001-32   

                                              FORNECEDOR  DOCUMENTO  \
39502  Papa Cucina Fast Food Italiano Comercio de Mas...      42458   
51961           Posto de Combust�ve QI 09Lago Sul Ltda  000005805   
52652                      GILBERTO PISELO DO NASCIMENTO     010/22   
52653                                      VIBRA ENERGIA  000971514   
52654                AMERICA VIAGENS E TURISMO LTDA - ME     WFMPQV   
...                                                  ...        ...   
69328                                      Exceller Tour     WIXHAI   
69329                                      Exceller Tour     WITOLM   
69330                                      Exceller Tour     THPKVQ   
69331                                      Exceller Tour     QNN9HX   
69332                                      Exceller Tour     WMQWBX   

            DATA                                       DETALHAMENTO  \
39502 2022-10-19                                    Alimenta��o   
51961 2022-12-27                                                NaN   
52652 2022-10-03  Despesa com aluguel de im�vel para uso do es...   
52653 2022-10-27  Despesa com aquisi��o de combust�vel par...   
52654 2022-10-04  Companhia A�rea: LATAM, Localizador: WFMPQV....   
...          ...                                                ...   
69328 2022-12-06  Companhia A�rea: LATAM, Localizador: WIXHAI....   
69329 2022-12-09  Companhia A�rea: GOL, Localizador: WITOLM. P...   
69330 2022-12-20  Companhia A�rea: TAM, Localizador: THPKVQ. P...   
69331 2022-12-21  Companhia A�rea: AZUL, Localizador: QNN9HX. ...   
69332 2022-12-30  Companhia A�rea: TAM, Localizador: WMQWBX. P...   

       VALOR_REEMBOLSADO  COD_DOCUMENTO  
39502             127.00        2174387  
51961             324.28        2175066  
52652            6000.00        2187616  
52653            4893.27        2189141  
52654            3575.00        2188975  
...                  ...            ...  
69328            2893.04        2191398  
69329            1180.19        2192272  
69330            2671.90        2192274  
69331            1334.31        2192244  
69332            2250.72        2193622  

[3600 rows x 11 columns]

In [99]:
mean_last_three_months = df_last_three_months.groupby('MES')['VALOR_REEMBOLSADO'].mean()
mean_last_three_months

MES
10.0    1634.281453
11.0    1771.556701
12.0    2954.796699
Name: VALOR_REEMBOLSADO, dtype: float64

Então podemos dizer que a média dos próximos 3 meses será 1634.281453, 1771.556701, 2954.796699

### Técnica 2: Prophet 


In [100]:
pip install Prophet

You should consider upgrading via the 'c:\Users\camil\AppData\Local\Programs\Python\Python310\python.exe -m pip install --upgrade pip' command.


In [101]:
from prophet import Prophet

In [102]:
df_prophet = pd.DataFrame() 

In [103]:
df_prophet['ds'] = combined_df['DATA']
df_prophet['y'] = combined_df['VALOR_REEMBOLSADO']
df_prophet = df_prophet.reset_index()

In [104]:
df_prophet=  df_prophet.drop(columns='index')


In [105]:
df_prophet = df_prophet.dropna()

In [106]:
m = Prophet()
m.fit(df_prophet)

10:51:16 - cmdstanpy - INFO - Chain [1] start processing
10:51:34 - cmdstanpy - INFO - Chain [1] done processing


In [107]:
future = m.make_future_dataframe(periods=90) # Predictions are then made on a dataframe with a column ds
future.tail()

ds
1641 2024-03-18
1642 2024-03-19
1643 2024-03-20
1644 2024-03-21
1645 2024-03-22

In [108]:
forecast = m.predict(future)
forecast[['ds', 'yhat', 'yhat_lower', 'yhat_upper']].tail()

ds         yhat   yhat_lower   yhat_upper
1641 2024-03-18  3014.712961  -646.968343  6948.404233
1642 2024-03-19  2852.745714 -1249.335804  6531.474365
1643 2024-03-20  2805.171396 -1165.830977  6851.903829
1644 2024-03-21  2778.712672 -1263.056810  6819.082443
1645 2024-03-22  2668.042328 -1467.246693  6771.596216

In [109]:
forecast

ds        trend   yhat_lower   yhat_upper  trend_lower  \
0    2000-12-08  3764.196000   273.162096  8111.417887  3764.196000   
1    2002-02-02  3588.715395  -493.959970  7210.669735  3588.715395   
2    2002-02-11  3584.964029   245.645399  7934.865769  3584.964029   
3    2002-02-19  3581.629480    -3.167349  7763.916277  3581.629480   
4    2002-03-14  3572.042654   -11.733391  7787.638745  3572.042654   
...         ...          ...          ...          ...          ...   
1641 2024-03-18  2523.895195  -646.968343  6948.404233  2523.346676   
1642 2024-03-19  2525.435751 -1249.335804  6531.474365  2524.809681   
1643 2024-03-20  2526.976307 -1165.830977  6851.903829  2526.321963   
1644 2024-03-21  2528.516863 -1263.056810  6819.082443  2527.834572   
1645 2024-03-22  2530.057419 -1467.246693  6771.596216  2529.310902   

      trend_upper  additive_terms  additive_terms_lower  additive_terms_upper  \
0     3764.196000      484.624312            484.624312            484.624312   
1     3588.715395     -415.539272           -415.539272           -415.539272   
2     3584.964029      319.022424            319.022424            319.022424   
3     3581.629480       95.852599             95.852599             95.852599   
4     3572.042654      282.309497            282.309497            282.309497   
...           ...             ...                   ...                   ...   
1641  2524.514420      490.817766            490.817766            490.817766   
1642  2526.079101      327.309962            327.309962            327.309962   
1643  2527.681076      278.195089            278.195089            278.195089   
1644  2529.297497      250.195809            250.195809            250.195809   
1645  2530.927898      137.984908            137.984908            137.984908   

          weekly  weekly_lower  weekly_upper      yearly  yearly_lower  \
0      36.246019     36.246019     36.246019  448.378293    448.378293   
1    -465.878872   -465.878872   -465.878872   50.339600     50.339600   
2     362.742183    362.742183    362.742183  -43.719760    -43.719760   
3     205.091148    205.091148    205.091148 -109.238549   -109.238549   
4     141.358560    141.358560    141.358560  140.950937    140.950937   
...          ...           ...           ...         ...           ...   
1641  362.742183    362.742183    362.742183  128.075583    128.075583   
1642  205.091148    205.091148    205.091148  122.218815    122.218815   
1643  162.461586    162.461586    162.461586  115.733503    115.733503   
1644  141.358560    141.358560    141.358560  108.837249    108.837249   
1645   36.246019     36.246019     36.246019  101.738889    101.738889   

      yearly_upper  multiplicative_terms  multiplicative_terms_lower  \
0       448.378293                   0.0                         0.0   
1        50.339600                   0.0                         0.0   
2       -43.719760                   0.0                         0.0   
3      -109.238549                   0.0                         0.0   
4       140.950937                   0.0                         0.0   
...            ...                   ...                         ...   
1641    128.075583                   0.0                         0.0   
1642    122.218815                   0.0                         0.0   
1643    115.733503                   0.0                         0.0   
1644    108.837249                   0.0                         0.0   
1645    101.738889                   0.0                         0.0   

      multiplicative_terms_upper         yhat  
0                            0.0  4248.820312  
1                            0.0  3173.176123  
2                            0.0  3903.986452  
3                            0.0  3677.482079  
4                            0.0  3854.352151  
...                          ...          ...  
1641                         0.0  3014.712961  
1642                         0.0  2852.745714  
1

In [110]:
forecast['MES'] = pd.to_datetime(forecast['ds'], format='%d/%m/%Y').dt.month
forecast['ANO'] = pd.to_datetime(forecast['ds'], format='%d/%m/%Y').dt.year

In [111]:
forecast

ds        trend   yhat_lower   yhat_upper  trend_lower  \
0    2000-12-08  3764.196000   273.162096  8111.417887  3764.196000   
1    2002-02-02  3588.715395  -493.959970  7210.669735  3588.715395   
2    2002-02-11  3584.964029   245.645399  7934.865769  3584.964029   
3    2002-02-19  3581.629480    -3.167349  7763.916277  3581.629480   
4    2002-03-14  3572.042654   -11.733391  7787.638745  3572.042654   
...         ...          ...          ...          ...          ...   
1641 2024-03-18  2523.895195  -646.968343  6948.404233  2523.346676   
1642 2024-03-19  2525.435751 -1249.335804  6531.474365  2524.809681   
1643 2024-03-20  2526.976307 -1165.830977  6851.903829  2526.321963   
1644 2024-03-21  2528.516863 -1263.056810  6819.082443  2527.834572   
1645 2024-03-22  2530.057419 -1467.246693  6771.596216  2529.310902   

      trend_upper  additive_terms  additive_terms_lower  additive_terms_upper  \
0     3764.196000      484.624312            484.624312            484.624312   
1     3588.715395     -415.539272           -415.539272           -415.539272   
2     3584.964029      319.022424            319.022424            319.022424   
3     3581.629480       95.852599             95.852599             95.852599   
4     3572.042654      282.309497            282.309497            282.309497   
...           ...             ...                   ...                   ...   
1641  2524.514420      490.817766            490.817766            490.817766   
1642  2526.079101      327.309962            327.309962            327.309962   
1643  2527.681076      278.195089            278.195089            278.195089   
1644  2529.297497      250.195809            250.195809            250.195809   
1645  2530.927898      137.984908            137.984908            137.984908   

          weekly  ...  weekly_upper      yearly  yearly_lower  yearly_upper  \
0      36.246019  ...     36.246019  448.378293    448.378293    448.378293   
1    -465.878872  ...   -465.878872   50.339600     50.339600     50.339600   
2     362.742183  ...    362.742183  -43.719760    -43.719760    -43.719760   
3     205.091148  ...    205.091148 -109.238549   -109.238549   -109.238549   
4     141.358560  ...    141.358560  140.950937    140.950937    140.950937   
...          ...  ...           ...         ...           ...           ...   
1641  362.742183  ...    362.742183  128.075583    128.075583    128.075583   
1642  205.091148  ...    205.091148  122.218815    122.218815    122.218815   
1643  162.461586  ...    162.461586  115.733503    115.733503    115.733503   
1644  141.358560  ...    141.358560  108.837249    108.837249    108.837249   
1645   36.246019  ...     36.246019  101.738889    101.738889    101.738889   

      multiplicative_terms  multiplicative_terms_lower  \
0                      0.0                         0.0   
1                      0.0                         0.0   
2                      0.0                         0.0   
3                      0.0                         0.0   
4                      0.0                         0.0   
...                    ...                         ...   
1641                   0.0                         0.0   
1642                   0.0                         0.0   
1643                   0.0                         0.0   
1644                   0.0                         0.0   
1645                   0.0                         0.0   

      multiplicative_terms_upper         yhat  MES   ANO  
0                            0.0  4248.820312   12  2000  
1                            0.0  3173.176123    2  2002  
2                            0.0  3903.986452    2  2002  
3                            0.0  3677.482079    2  2002  
4                            0.0  3854.352151    3  2002  
...                          ...          ...  ...   ...  
1641                         0.0  3014.712961    3  2024  
1642                         0.0  2852.745714    3  2024  
1643      

In [112]:
next_two = [2023,2024]
prophet_predict = forecast.loc[forecast['ANO'].isin(next_two)]

In [113]:
prophet_predict

ds        trend   yhat_lower   yhat_upper  trend_lower  \
1523 2023-01-01  1842.969415 -2094.429831  5876.026520  1842.969415   
1524 2023-01-02  1844.509971 -1328.027549  6667.322890  1844.509971   
1525 2023-01-03  1846.050527 -1545.242856  6263.083124  1846.050527   
1526 2023-01-04  1847.591083 -1719.406375  5984.087616  1847.591083   
1527 2023-01-05  1849.131639 -1791.493671  5866.056184  1849.131639   
...         ...          ...          ...          ...          ...   
1641 2024-03-18  2523.895195  -646.968343  6948.404233  2523.346676   
1642 2024-03-19  2525.435751 -1249.335804  6531.474365  2524.809681   
1643 2024-03-20  2526.976307 -1165.830977  6851.903829  2526.321963   
1644 2024-03-21  2528.516863 -1263.056810  6819.082443  2527.834572   
1645 2024-03-22  2530.057419 -1467.246693  6771.596216  2529.310902   

      trend_upper  additive_terms  additive_terms_lower  additive_terms_upper  \
1523  1842.969415      -38.504114            -38.504114            -38.504114   
1524  1844.509971      713.805277            713.805277            713.805277   
1525  1846.050527      503.783956            503.783956            503.783956   
1526  1847.591083      409.654964            409.654964            409.654964   
1527  1849.131639      338.677336            338.677336            338.677336   
...           ...             ...                   ...                   ...   
1641  2524.514420      490.817766            490.817766            490.817766   
1642  2526.079101      327.309962            327.309962            327.309962   
1643  2527.681076      278.195089            278.195089            278.195089   
1644  2529.297497      250.195809            250.195809            250.195809   
1645  2530.927898      137.984908            137.984908            137.984908   

          weekly  ...  weekly_upper      yearly  yearly_lower  yearly_upper  \
1523 -442.020624  ...   -442.020624  403.516510    403.516510    403.516510   
1524  362.742183  ...    362.742183  351.063094    351.063094    351.063094   
1525  205.091148  ...    205.091148  298.692808    298.692808    298.692808   
1526  162.461586  ...    162.461586  247.193378    247.193378    247.193378   
1527  141.358560  ...    141.358560  197.318776    197.318776    197.318776   
...          ...  ...           ...         ...           ...           ...   
1641  362.742183  ...    362.742183  128.075583    128.075583    128.075583   
1642  205.091148  ...    205.091148  122.218815    122.218815    122.218815   
1643  162.461586  ...    162.461586  115.733503    115.733503    115.733503   
1644  141.358560  ...    141.358560  108.837249    108.837249    108.837249   
1645   36.246019  ...     36.246019  101.738889    101.738889    101.738889   

      multiplicative_terms  multiplicative_terms_lower  \
1523                   0.0                         0.0   
1524                   0.0                         0.0   
1525                   0.0                         0.0   
1526                   0.0                         0.0   
1527                   0.0                         0.0   
...                    ...                         ...   
1641                   0.0                         0.0   
1642                   0.0                         0.0   
1643                   0.0                         0.0   
1644                   0.0                         0.0   
1645                   0.0                         0.0   

      multiplicative_terms_upper         yhat  MES   ANO  
1523                         0.0  1804.465301    1  2023  
1524                         0.0  2558.315248    1  2023  
1525                         0.0  2349.834483    1  2023  
1526                         0.0  2257.246047    1  2023  
1527                         0.0  2187.808976    1  2023  
...                          ...          ...  ...   ...  
1641                         0.0  3014.712961    3  2024  
1642                         0.0  2852.745714    3  2024  
1643      

In [114]:
sum_next_two_years = prophet_predict.groupby('ANO')['yhat'].agg('sum')
sum_next_two_years

ANO
2023     93512.361382
2024    205461.318231
Name: yhat, dtype: float64

### Técnica 3: Regressão Linear

In [115]:
pip install sklearn

Note: you may need to restart the kernel to use updated packages.


You should consider upgrading via the 'c:\Users\camil\AppData\Local\Programs\Python\Python310\python.exe -m pip install --upgrade pip' command.


In [116]:
from sklearn.linear_model import LinearRegression # importa o modelo
from sklearn.model_selection import train_test_split

In [128]:
X = combined_df['DATA']
y = combined_df['VALOR_REEMBOLSADO']

In [129]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
